htdocs/contrat/list.php

```diff
@@ -202,6 +210,7 @@
 $sql.= ' s.rowid as socid, s.nom as name, s.town, s.zip, s.fk_pays, s.client, s.code_client,';
 $sql.= " typent.code as typent_code,";
 $sql.= " state.code_departement as state_code, state.nom as state_name,";
+$sql.= " MIN(CASE WHEN cd.statut=4 THEN cd.date_fin_validite END) as first_date_fin_validite,";
```

Comment:
To have sql condition portable, can you use 
$db->ifsql   method 
instead of CASE WHEN ?